### PR TITLE
Better icon system

### PR DESCRIPTION
**Pull request by 1egoman**
Reasoning:
- Giving each icon its own component means that it's much easier to understand which icon is which, rather than looking at non-self-describing unicode characters in the icon font.
- Each component renders each icon using svg, which is [pretty much supported everywhere now](https://caniuse.com/#feat=svg). Also, this means that doing things such as changing the icon's color can be done by passing a prop to the component.

screen shot 2018-05-01 at 2 27 22 pm

DEN-4309

### CODE REVIEW

File: src/components/icons/index.js [hunk 1]
```diff
@@ -0,0 +1,860 @@
...
+              <path d="M8,10 L18,10" id="Stroke-7" stroke={color} strokeWidth="1.5" strokeLinejoin="round"
+              />
+              <path d="M2,15 L6,15" id="Stroke-9" stroke={color} strokeWidth="1.5" strokeLinejoin="round"
+              />
+              <path d="M8,15 L18,15" id="Stroke-11" stroke={color} strokeWidth="1.5"
+              strokeLinejoin="round" />
+          </g>
+      </g>
+  </svg>,
+  IconCheveronDownCircle: ({color, width, height}) => <svg width={width || "18"} height={height || "18"} viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
```
Comment (guscost) [inline src/components/icons/index.js]: "Chevron" is the modern spelling.

Comment (1egoman) [reply src/components/icons/index.js]: :) I can't spell. Will fix.

**Comment by 1egoman** [comment]
> How do we maintain the icons? Is there a process to update the SVG paths or add/remove icons?

This would be a great thing to document! My process for adding them in here was:
1. Find `densicons.sketch` in the Density google drive account.
2. In the sketch sidebar, right click on each icon and select `Copy CVG Code`.
3. Paste the resulting svg into the right pane of this tool: https://svg2jsx.herokuapp.com. Copy the output from the right pane.
4. Add the icon to the `ICONS` object in `components/icons/index.js`, where the key is the component name and the value is a function that accepts any props that can be passed to the component and returns the resulting jsx from the svg2jsx tool (parameterized with any props that would be desired)

Let me write that up in a bit more official manor and add it into the icon component's readme.

**Comment by 1egoman** [comment]
@guscost Does this address your concern? https://github.com/DensityCo/ui/tree/better-icon-system-DEN-4309/src/components/icons#density-ui-icons

**Pull request merged**